x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java

```diff
@@ -61,6 +62,10 @@ public DataType dataType() {
 
     @Override
     public String innerName() {
+        if (!percent.foldable()) {
```

Comment:
The check happens too late when the Percentile is going to be executed. It's better to move it up-front in the `resolveType` since that's where we typically check the arguments.
This would also avoid having to create an exception and instead return a resolution message.